backend/python/app/services/implementations/story_service.py

```diff
@@ -218,54 +218,93 @@ def assign_user_as_reviewer(self, user, story_translation):
+
+        if story_translation_stage == "TRANSLATE":
+            try:
+                if not story_translation:
+                    raise Exception(
+                        "story_translation_content_id {id} not found".format(
+                            id=story_translation_content.id
+                        )
+                    )
+
+                story_translation.translation_content = (
+                    story_translation_content.translation_content
+                )
+                db.session.commit()
+            except Exception as error:
+                reason = getattr(error, "message", None)
+                self.logger.error(
+                    "Failed to update story translation content. Reason = {reason}".format(
+                        reason=(reason if reason else str(error))
                     )
                 )
+                raise error
 
-            story_translation.translation_content = (
-                story_translation_content.translation_content
+            return StoryTranslationContentResponseDTO(
+                story_translation_content.id,
+                story_translation.line_index,
+                story_translation_content.translation_content,
             )
-            db.session.commit()
-        except Exception as error:
-            reason = getattr(error, "message", None)
-            self.logger.error(
-                "Failed to update story translation content. Reason = {reason}".format(
-                    reason=(reason if reason else str(error))
-                )
+        elif story_translation_stage == "REVIEW":
+            raise Exception(
+                "Story translation contents cannot be changed while the story is being reviewed."
             )
-            raise error
-
-        return StoryTranslationContentResponseDTO(
-            story_translation_content.id,
-            story_translation.line_index,
-            story_translation_content.translation_content,
-        )
+        elif story_translation_stage == "PUBLISH":
+            raise Exception(
+                "Story translation contents cannot be changed after the story has been published."
+            )
+        else:
+            raise Exception("Story translation contents cannot be changed right now.")
 
     def update_story_translation_contents(self, story_translation_contents):
-        try:
-            # TODO: return lineIndex too
-            db.session.bulk_update_mappings(
-                StoryTranslationContent, story_translation_contents
-            )
-            db.session.commit()
-            return story_translation_contents
-        except Exception as error:
-            reason = getattr(error, "message", None)
-            self.logger.error(
-                "Failed to update story translation content. Reason = {reason}".format(
-                    reason=(reason if reason else str(error))
+        story_translation_content = StoryTranslationContent.query.filter_by(
```

Comment:
ditto comment about wrapping this in a try statement. You should always wrap any database interactions with try statements, because they could always fall through if the database has issues!